src/fft.jl

```diff
@@ -0,0 +1,576 @@
+                overlap_y = intersect(src_domain.indexes[2], dst_domain.indexes[2])
+                overlap_z = intersect(src_domain.indexes[3], dst_domain.indexes[3])
+                
+                if !isempty(overlap_x) && !isempty(overlap_y) && !isempty(overlap_z)
+                    if dst_chunk.handle.rank == rank
+                        # Local copy
+                        dst_data = fetch(dst_chunk)
+                        src_indices = (
+                            overlap_x .- first(src_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(src_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(src_domain.indexes[3]) .+ 1
+                        )
+                        dst_indices = (
+                            overlap_x .- first(dst_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(dst_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(dst_domain.indexes[3]) .+ 1
+                        )
+                        dst_data[dst_indices...] .= view(src_data, src_indices...)
+                    else
+                        # Remote send using views!!!
+                        src_indices = (
+                            overlap_x .- first(src_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(src_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(src_domain.indexes[3]) .+ 1
+                        )
+                        send_data = view(src_data, src_indices...)
+                        
+                        tag = src_idx * nranks + dst_idx
+                        req = MPI.Isend(send_data, dst_chunk.handle.rank, tag, comm)
+                        push!(reqs, req)
+                    end
+                end
+            end
+        end
+    end
+    
+    if !isempty(reqs)
+        MPI.Waitall(reqs)
+    end
+    
+    MPI.Barrier(comm)
+end
+
+
+function fft(A::DArray{T,3}, B::DArray{T,3}, C::DArray{T,3}, transforms, dims) where T
+    A_parts = A.chunks
+    B_parts = B.chunks
+    C_parts = C.chunks
+    
+    Dagger.spawn_datadeps() do
+        for idx in eachindex(A_parts)
+            Dagger.@spawn name="apply_fft!(dim 1)[$idx]" apply_fft!(InOut(A_parts[idx]), In(transforms[1]), In(dims[1]))
+        end
+    end
+
+    redistribute_x_to_y!(B, A)
+
+    Dagger.spawn_datadeps() do
+        for idx in eachindex(B_parts)
+            Dagger.@spawn name="apply_fft!(dim 2)[$idx]" apply_fft!(InOut(B_parts[idx]), In(transforms[2]), In(dims[2]))
+        end
+    end
+
+    redistribute_y_to_z!(C, B)
+    Dagger.spawn_datadeps() do
+        for idx in eachindex(C_parts)
+            Dagger.@spawn name="apply_fft!(dim 3)[$idx]" apply_fft!(InOut(C_parts[idx]), In(transforms[3]), In(dims[3]))
+        end
+    end
+
+    return C
+end
+
+function fft(A::DArray{T,3}, B::DArray{T,3}, transforms, dims, decomp::Decomposition = Slab()) where T 
```

Comment:
To select the decomposition, I think we should make this a kwarg (like `fft(A, B; dims, decomp)`), and add it to all methods.

Reply:
```suggestion
function AbstractFFTs.fft(A::DArray{T,3}; dims, decomp::Decomposition = Slab()) where T 
```